### PR TITLE
Implement membership file download

### PR DESCRIPTION
**Pull request by rushtong**
Update tsv formatter to handle membership files.
Update export actor to zip up membership and entity file for download.

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/firecloud/core/ExportEntitiesByType.scala [hunk 1]
```diff
@@ -34,27 +38,72 @@ class ExportEntitiesByTypeActor(requestContext: RequestContext) extends Actor wi
   val log = Logging(system, getClass)
 
   override def receive: Receive = {
-    case ProcessEntities(encodedUrl: String, filename: String, entityType: String) =>
+    case ProcessEntities(baseEntitydUrl: String, filename: String, entityType: String) =>
```
Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/firecloud/core/ExportEntitiesByType.scala]: typo in variable name

File: src/main/scala/org/broadinstitute/dsde/firecloud/utils/TSVFormatter.scala [hunk 2]
```diff
@@ -2,14 +2,34 @@ package org.broadinstitute.dsde.firecloud.utils
...
+import spray.json.JsValue
+
 object TSVFormatter {
 
-  def makeTsvString(entities: List[EntityWithType], entityType: String) = {
+  def makeMembershipTsvString(entities: List[EntityWithType], entityType: String): String = {
+    val headers: immutable.IndexedSeq[String] = immutable.IndexedSeq("membership:" + entityType + "_id", entityType.replace("_set", "_id"))
+    val rows: List[IndexedSeq[String]] = entities filter { _.entityType == entityType } flatMap { entity =>
+      entity.attributes.getOrElse(Map.empty) flatMap { m =>
+        m._2.convertTo[List[JsValue]] map { jsValue =>
```
Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/firecloud/utils/TSVFormatter.scala:18]: Could you explain the logic here?

**Comment by jmthibault79** [comment]
Verified locally that it works.  A few comments.

File: src/main/scala/org/broadinstitute/dsde/firecloud/utils/TSVFormatter.scala (hunk 2 above)
Comment (rushtong) [inline src/main/scala/org/broadinstitute/dsde/firecloud/utils/TSVFormatter.scala:18]: The json attributes in a membership object (i.e. pair_set, sample_set, participant_set) represent rows as opposed to the root types, where the attributes represent columns. So, for each set attribute, we need to get all of the elements out of it and turn each one into a row. Here, `m._2` is the attribute list. We unmarshal that into a list and then work with each value. 

Here is a pair_set example:

```
[{
  "name": "pair_set_1",
  "entityType": "pair_set",
  "attributes": {
    "pairs": [{
      "entityType": "pair",
      "entityName": "1"
    }, {
      "entityType": "pair",
      "entityName": "2"
    }]
  }
}]
```

File: src/test/scala/org/broadinstitute/dsde/firecloud/utils/TSVFormatterSpec.scala [hunk 3]
```diff
@@ -14,91 +15,104 @@ class TSVFormatterSpec extends FreeSpec with ScalaFutures with Matchers with Ins
...
-          "control_sample_id" -> """{"entityType": "sample", "entityName": "HCC1143_BL"}""".parseJson,
-          "participant_id" -> """{"entityType":"participant_set","entityName":"subject_HCC1143"}""".parseJson,
-          "gender" -> "male".toJson,
-          "header_1" -> MockUtils.randomAlpha().toJson,
-          "header_2" -> MockUtils.randomAlpha().toJson
-        )
+    "Participant tests should pass for" - {
+
+      "Entity Data" in {
+        val participantSetAtts = {
```
Comment (jmthibault79) [inline src/test/scala/org/broadinstitute/dsde/firecloud/utils/TSVFormatterSpec.scala]: This should represent attributes for a single participant, but is named like a participant set, and contains attributes appropriate for pairs.

Comment (rushtong) [inline src/test/scala/org/broadinstitute/dsde/firecloud/utils/TSVFormatterSpec.scala]: It's test data - the values don't mean anything. I'll update it though.

File: src/main/scala/org/broadinstitute/dsde/firecloud/utils/TSVFormatter.scala (hunk 2 above)
Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/firecloud/utils/TSVFormatter.scala:18]: ok thanks

File: src/test/scala/org/broadinstitute/dsde/firecloud/utils/TSVFormatterSpec.scala (hunk 3 above)
Comment (jmthibault79) [inline src/test/scala/org/broadinstitute/dsde/firecloud/utils/TSVFormatterSpec.scala]: Thanks, I appreciate it.  My branch makes these tests more strict, so this helps me.

**Comment by jmthibault79** [comment]
I'm happy.  Rebase and squash and we're good to go.

**Comment by rushtong** [comment]
Merging.

**Pull request merged**